plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/grok/GrokTests.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.grok;
+
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matcher;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.Object;
+import java.lang.String;
+import java.util.*;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+
+public class GrokTests extends ESTestCase {
+
+    private Grok baseGrok;
+
+    @Before
+    public void setup() {
+        baseGrok = new Grok();
+        //try {
```

Comment:
I think you still can add the named patterns, since the files are still on the class path?

```java
grok.loadFromStream(getClass().getResourceAsStream("/config/ingest/patterns/aws"));
grok.loadFromStream(getClass().getResourceAsStream("/config/ingest/patterns/bacula"));
// next file
```